pulsar-client/src/main/java/org/apache/pulsar/client/impl/NegativeAcksTracker.java

```diff
@@ -95,6 +105,40 @@ public synchronized void add(MessageId messageId) {
         }
     }
 
+    public synchronized void add(Message<?> message) {
+        if (negativeAckRedeliveryBackoff == null) {
+            add(message.getMessageId());
+            return;
+        }
+        add(message.getMessageId(), message.getRedeliveryCount());
+    }
+
+    private synchronized void add(MessageId messageId, int redeliveryCount) {
+        if (messageId instanceof TopicMessageIdImpl) {
```

Comment:
The `TopicMessageIdImpl` will not reach here? The NegativeAcksTracker maintained in the internal consumers

Reply:
If this message from MultiTopicsConsumerImpl, messageId instance of TopicMessageIdImpl.